OpenDreamRuntime/DreamRuntime.cs

```diff
@@ -0,0 +1,239 @@
+        public Dictionary<DreamList, DreamObject> UnderlaysListToAtom = new();
+        public List<DreamObject> Mobs = new ();
+        public DreamList WorldContentsList;
+
+        public DreamRuntime(DreamServer server, string executablePath) {
+			MainThread = Thread.CurrentThread;
+            Server = server;
+
+            // Something is doing something fucky with relative dirs, somewhere
+            executablePath = Path.GetFullPath(executablePath);
+
+            ResourceManager = new DreamResourceManager(Path.GetDirectoryName(executablePath));
+
+            // This initialization isn't great
+            ObjectTree = new(this);
+            StateManager = new(this);
+
+			_taskScheduler = new();
+			TaskFactory = new TaskFactory(_taskScheduler);
+
+            StateManager.DeltaStateFinalized += OnDeltaStateFinalized;
+            Server.DreamConnectionRequest += OnDreamConnectionRequest;
+
+            TickStartTime = new DateTimeOffset(DateTime.Now).ToUnixTimeMilliseconds();
+        
+			CompiledJson = LoadCompiledJson(executablePath);
+			if (CompiledJson == null) {
+				throw new InvalidOperationException();
+			}
+
+			_clientInterface = LoadInterface(CompiledJson.Interface);
+			if (_clientInterface == null) {
+				throw new InvalidOperationException();
+			}
+
+            ObjectTree.LoadFromJson(CompiledJson.RootObject);
+            ObjectTree.SetMetaObject(DreamPath.Root, new DreamMetaObjectRoot(this));
+            ObjectTree.SetMetaObject(DreamPath.List, new DreamMetaObjectList(this));
+            ObjectTree.SetMetaObject(DreamPath.Sound, new DreamMetaObjectSound(this));
+            ObjectTree.SetMetaObject(DreamPath.Image, new DreamMetaObjectImage(this));
+            ObjectTree.SetMetaObject(DreamPath.World, new DreamMetaObjectWorld(this));
+            ObjectTree.SetMetaObject(DreamPath.Client, new DreamMetaObjectClient(this));
+            ObjectTree.SetMetaObject(DreamPath.Datum, new DreamMetaObjectDatum(this));
+            ObjectTree.SetMetaObject(DreamPath.Regex, new DreamMetaObjectRegex(this));
+            ObjectTree.SetMetaObject(DreamPath.Atom, new DreamMetaObjectAtom(this));
+            ObjectTree.SetMetaObject(DreamPath.Area, new DreamMetaObjectArea(this));
+            ObjectTree.SetMetaObject(DreamPath.Turf, new DreamMetaObjectTurf(this));
+            ObjectTree.SetMetaObject(DreamPath.Movable, new DreamMetaObjectMovable(this));
+            ObjectTree.SetMetaObject(DreamPath.Mob, new DreamMetaObjectMob(this));
+            DreamProcNative.SetupNativeProcs(ObjectTree);
+
+            ListDefinition = ObjectTree.GetObjectDefinitionFromPath(DreamPath.List);
+
+            WorldInstance = ObjectTree.CreateObject(DreamPath.World);
+            ObjectTree.GetObjectDefinitionFromPath(DreamPath.Root).GlobalVariables["world"].Value = new DreamValue(WorldInstance);
+
+            Map = new DreamMap(this);
+            Map.LoadMap(CompiledJson.Maps[0]);

```

Comment:
Loading the map should be done after the global init proc is called, and before `/world.New()` is called.

Reply:
Ah, I messed around with this when porting the Main from OpenDreamServer and must have missed this bit when I tried to keep the ordering the same.